src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1926,4 +1942,27 @@ public static String getVersionName() {
         // value
         return "1.1.12";
     }
+
+    /**
+     * A {@link Future}  whose result can be set by a {@link #set(Object)} or {@link #setException(Throwable)}
+     */
+    final static class SettableFuture<V> extends FutureTask<V> {
```

Comment:
The problem is that we currently don't notify user that some RuntimeException happened (callback won't be invoked if this happens). So caller doesn't know when to call get() function.
Moreover I see that all our async calls don't check Future at all, so RuntimeExceptions got swallowed. 
I'm actually open to any proposals how to fix it...